### PR TITLE
Work around missing cmake config files in PyPI tbb-devel, require TBB for pip packages

### PR DESCRIPTION
**Pull request by SimonHeybrock**
See https://github.com/oneapi-src/oneTBB/issues/682 for upstream issue. This PR is a workaround.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -17,3 +17,5 @@ test-command = "pytest {project}/tests"
 # PyPy builds fail with an error about ninja, probably https://github.com/pypa/setuptools/issues/2912
 # tbb-devel not available for musllinux
 skip = ["pp*", "*musllinux*"]
+build = ["cp39-manylinux_x86_64", "cp38-macosx_x86_64", "cp39-win_amd64"]
```
Comment (jl-wynen) [inline pyproject.toml:20]: What does the `cp` prefix mean? And why is it different for MacOS?

**Review by jl-wynen — COMMENTED** [review]
Where does TBB come from in the pip build? You removed it from `setup.py`. Is it preinstalled or does CMake find the version that was installed by conda?

**Comment by SimonHeybrock** [comment]
> Where does TBB come from in the pip build? You removed it from `setup.py`. Is it preinstalled or does CMake find the version that was installed by conda?

`pyproject.toml` requires `tbb-devel`.

The reason why `tbb` is removed from the *install* requirements is that we now include `libtbb.so` inside the scipp package.

File: pyproject.toml (hunk 1 above)
Comment (SimonHeybrock) [reply pyproject.toml:20]: Just for testing, made only one package per OS, this will be removed. 3.8 on mac, because my Laptop has that Python version.

**Comment by SimonHeybrock** [comment]
CI has passed before squashing commit, merging.

**Pull request merged**